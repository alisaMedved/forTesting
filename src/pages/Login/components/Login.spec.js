import React from 'react'
import { shallow } from 'enzyme'
import Login from './Login'

describe('Login', () => {
  const mockLogin = jest.fn()

  const initialState = {
    email: '',
    password: '',
    redirectToPreviousRoute: false,
  }

  const props = {
    location: { state: { from: '/' } },
    logIn: mockLogin,
    errorMsg: null,
  }
  const login = shallow(<Login {...props} />)

  it('renders properly', () => {
    expect(login).toMatchSnapshot()
  })

  it('initialize Login with initial state', () => {
    expect(login.state()).toEqual(initialState)
  })

  describe('Form handlers', () => {
    describe('when typing into email input', () => {
      const email = 'max@test.com'

      beforeEach(() => {
        login.find('[data-field-name="email"]').simulate('change', {
          currentTarget: {
            value: email,
            dataset: {
              fieldName: 'email',
            },
          },
        })
      })

      it('updates email field in state', () => {
        expect(login.state().email).toEqual(email)
      })
    })
    describe('when typing into password input', () => {
      const password = '11'

      beforeEach(() => {
        login.find('[data-field-name="password"]').simulate('change', {
          currentTarget: {
            value: password,
            dataset: {
              fieldName: 'password',
            },
          },
        })
      })

      it('updates password field in state', () => {
        expect(login.state().password).toEqual(password)
      })
    })

    describe('when submiting the form', () => {
      beforeEach(() => {
        login.find('form').simulate('submit', {
          preventDefault: () => {},
        })
      })

      it('calls the props.logIn', () => {
        /* expect(mockLogin).toHaveBeenCalledWith(
          {
            email: 'max@test.com',
            password: '11',
          }
        ) */
        expect(mockLogin).toHaveBeenCalledTimes(1)
      })
    })
    describe('when clicking the Login button', () => {
      login.find('button').simulate('click', {
        preventDefault: () => {},
      })

      it('calls the props.logIn', () => {
        expect(mockLogin).toHaveBeenCalledTimes(1)
      })
    })

    afterAll(() => {
      login.setState(initialState)
    })
  })
})
