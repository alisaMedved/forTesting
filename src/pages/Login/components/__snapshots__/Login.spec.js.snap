// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    errorMsg={null}
    location={
        Object {
            "state": Object {
              "from": "/",
            },
          }
    }
    logIn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <form
          onSubmit={[Function]}
>
          <input
                    data-field-name="email"
                    onChange={[Function]}
                    placeholder="Имя"
                    type="text"
                    value=""
          />
          <input
                    data-field-name="password"
                    onChange={[Function]}
                    placeholder="Пароль"
                    type="text"
                    value=""
          />
          <button
                    type="submit"
          >
                    Log in
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              data-field-name="email"
              onChange={[Function]}
              placeholder="Имя"
              type="text"
              value=""
/>,
            <input
              data-field-name="password"
              onChange={[Function]}
              placeholder="Пароль"
              type="text"
              value=""
/>,
            <button
              type="submit"
>
              Log in
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data-field-name": "email",
              "onChange": [Function],
              "placeholder": "Имя",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data-field-name": "password",
              "onChange": [Function],
              "placeholder": "Пароль",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log in",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Log in",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <form
            onSubmit={[Function]}
>
            <input
                        data-field-name="email"
                        onChange={[Function]}
                        placeholder="Имя"
                        type="text"
                        value=""
            />
            <input
                        data-field-name="password"
                        onChange={[Function]}
                        placeholder="Пароль"
                        type="text"
                        value=""
            />
            <button
                        type="submit"
            >
                        Log in
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                data-field-name="email"
                onChange={[Function]}
                placeholder="Имя"
                type="text"
                value=""
/>,
              <input
                data-field-name="password"
                onChange={[Function]}
                placeholder="Пароль"
                type="text"
                value=""
/>,
              <button
                type="submit"
>
                Log in
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "data-field-name": "email",
                "onChange": [Function],
                "placeholder": "Имя",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "data-field-name": "password",
                "onChange": [Function],
                "placeholder": "Пароль",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log in",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Log in",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
